common/ellipses.js

```diff
@@ -181,36 +181,56 @@
+                                scope.hideContent = true;
+                                containsOverflow = true;
+                                scope.maxHeightStyle =  maxHeightStyle;
+                            } else {
+                                scope.overflow[index] = false;
+                            }
+                        }
+
+                        // Resizerow is called whenever there is a change in rowValues model
+                        // It iterates over all the td elements and extracts image and iframes from it
+                        // After which it binds onload event to adjust height
+                        // It also calls updateHeight for the same td, for any oveflown textual content
                         var resizeRow = function() {
-                            if (containsOverflow == false && timerCount ++ < 500) {
+                            if (containsOverflow == false) {
 
+                                //Iterate over table data in the row
                                 for (var i = 0; i < element[0].children.length; i++) {
-                                    var height = element[0].children[i].children[0].clientHeight;
-                                    if (height > maxHeight) {
-                                        scope.overflow[i] = true;
-                                        scope.hideContent = true;
-                                        containsOverflow = true;
-                                        scope.maxHeightStyle =  maxHeightStyle;
-                                    } else {
-                                        scope.overflow[i] = false;
-                                    }
+
+                                    // Get all images and iframes inside the td  
+                                    var imagesAndIframes = UiUtils.getImageAndIframes(element[0].children[i]);
+
+                                    // Bind onload event and updateheight for particular td index
+                                    imagesAndIframes.forEach(function(el) {
+                                        var index = i;
+                                        el.onload = function() {
+                                            updateHeight(index, el);
+                                        };
+                                    });
+
+                                    updateHeight(i, element[0].children[i].children[0]);
                                 }
-                                $timeout(function() {
-                                    resizeRow();
-                                }, 50);
                             }
                         };
 
+                        // Watch for change in rowValues, this is useful in case of pagination
+                        // As Angular just changes the content and doesnot destroys elements
                         scope.$watchCollection('rowValues', function (v) {
```

Comment:
So basically, when the directive is called, the html is still not displayed. So we need to add a timeout to call resizeRow. 

As Javascript uses event queue. The `resizeRow` function is called after the html is rendered by the directive because of the timeout. Even if I make the timeout to 1, it will still work.